crates/core/src/ast_node.rs

```diff
@@ -0,0 +1,47 @@
+use tree_sitter::Node;
+
+/// Represents an AST node and offers convenient AST-specific functionality.
+///
+/// This trait should be free from dependencies on TreeSitter.
+pub trait AstNode: Sized {
+    /// Returns the next node, ignoring trivia such as whitespace.
+    fn next_non_trivia_node(&self) -> Option<Self>;
+
+    /// Returns the previous node, ignoring trivia such as whitespace.
+    fn previous_non_trivia_node(&self) -> Option<Self>;
+}
```

Comment:
Not blocking for this pr,
Ideally this trait should also implement the trait from tree-sitter-traverse. alternatively we could excise tree-sitter-traverse and have this trait replace that as the crate is only one file. tree-sitter-traverse trait is not dependent on tree-sitter